File: 백준/Bronze/1009. 분산처리/분산처리.swift
func BOJ_1009(_ T: Int) {
    var result = ""
    for _ in 0..<T {
        let ab = readLine()!.split(separator: " ").map { Int($0)! }, base = ab[0] % 10
        var temp = 1
        for _ in 1...ab[1] {
            if temp == 0 { break }
            temp = (temp * base) % 10
        }
        result.append("\(temp == 0 ? 10 : temp)\n")
    }
    print(result)
}
BOJ_1009(Int(readLine()!)!)

File: 백준/Bronze/1009. 분산처리/분산처리.py
import sys
input = sys.stdin.readline
t = int(input())
result = []
for _ in range(t):
    a, b = map(int, input().split())
    last = pow(a, b, 10)
    result.append(str(10 if last == 0 else last))
print('\n'.join(result))